nis/src/main/java/org/nem/nis/dao/retrievers/MosaicSupplyRetriever.java

```diff
@@ -15,19 +16,38 @@
  * Class for for retrieving mosaic supplies.
  */
 public class MosaicSupplyRetriever {
+	final private int namespaceLifetime;
+
+	/**
+	 * Creates a retriever with default settings.
+	 */
+	public MosaicSupplyRetriever() {
+		this(NemGlobals.getBlockChainConfiguration().getEstimatedBlocksPerDay() * (365 + 30 + 1));
```

Comment:
Is the plus one needed if the grace period is 30 days? 🤔

Reply:
i think so, i matched what the pruning observer is doing here: https://github.com/NemProject/nem/blob/dev/nis/src/main/java/org/nem/nis/secret/pruning/NamespaceCachePruningObserver.java#L24